splice_spark2/src/main/spark2.2/com/splicemachine/spark2/splicemachine/SplicemachineContext.scala

```diff
@@ -595,11 +595,13 @@ class SplicemachineContext(options: Map[String, String]) extends Serializable {
     )
   
   /** Schema string built from JDBC metadata. */
-  def schemaString(schemaTableName: String): String =
+  def schemaString(schemaTableName: String, schema: StructType = new StructType()): String =
     SpliceJDBCUtil.retrieveColumnInfo(
       new JDBCOptions(Map(
         JDBCOptions.JDBC_URL -> url,
         JDBCOptions.JDBC_TABLE_NAME -> schemaTableName))
+    ).filter(
+      col => schema.isEmpty || schema.exists( field => field.name.toUpperCase.equals( col(0).toUpperCase ) )
```

Comment:
Does this work for quoted columns? 

`create table a ("notEQUAL" int, "NOTequal" int);`

Reply:
Verified now that it works for unquoted columns but not quoted columns.  I’ll file a jira.

Reply:
Wrote https://splicemachine.atlassian.net/browse/DB-9581